Makefile

```diff
@@ -715,49 +748,72 @@ clean-generated-conversions: ## Remove files generated by conversion-gen from th
 
 ##@ hack/tools:
 
-controller-gen: $(CONTROLLER_GEN) ## Build a local copy of controller-gen
-conversion-gen: $(CONVERSION_GEN) ## Build a local copy of conversion-gen
-conversion-verifier: $(CONVERSION_VERIFIER) ## Build a local copy of conversion-verifier
-gotestsum: $(GOTESTSUM) ## Build a local copy of gotestsum
-go-apidiff: $(GO_APIDIFF) ## Build a local copy of apidiff
-golangci-lint: $(GOLANGCI_LINT) ## Build a local copy of golangci-lint
-envsubst: $(ENVSUBST) ## Build a local copy of envsubst
-kustomize: $(KUSTOMIZE) ## Build a local copy of kustomize
-setup-envtest: $(SETUP_ENVTEST) ## Build a local copy of setup-envtest
-tilt-prepare: $(TILT_PREPARE) ## Build a local copy of tilt-prepare
-kpromo: $(KPROMO)  ## Build a local copy of kpromo
+.PHONY: $(CONTROLLER_GEN_BIN)
+$(CONTROLLER_GEN_BIN): $(CONTROLLER_GEN) ## Build a local copy of controller-gen.
 
-$(SETUP_ENVTEST): $(TOOLS_DIR)/go.mod # Build setup-envtest from tools folder.
-	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/setup-envtest sigs.k8s.io/controller-runtime/tools/setup-envtest
+## We are forcing a rebuilt of conversion-gen via PHONY so that we're always using an up-to-date version.
+## We can't use a versioned name for the binary, because that would be reflected in generated files.
```

Comment:
This should be above: l.789 instead: (+ we have to add the PHONY line there)
```bash
## We are forcing a rebuilt of conversion-gen via PHONY so that we're always using an up-to-date version.
## We can't use a versioned name for the binary, because that would be reflected in generated files.
.PHONY: $(CONVERSION_GEN)
$(CONVERSION_GEN): # Build conversion-gen from tools folder.
```

Reply:
But overall looks like we solved it for top-level. Now it's only about fixing it in test/infrastructure/docker too.